x-pack/plugins/apm/server/lib/helpers/transaction_group_query.ts

```diff
@@ -86,17 +87,18 @@ export function prepareTransactionGroups({
     const averageResponseTime = bucket.avg.value;
     const transactionsPerMinute = bucket.doc_count / minutes;
     const impact = Math.round(averageResponseTime * transactionsPerMinute);
-    const sample = oc(bucket).sample.hits.hits[0]._source();
+    const sample = bucket.sample.hits.hits[0]._source;
 
     return {
       name: bucket.key,
-      serviceName: oc(sample).context.service.name('n/a'),
-      id: oc(sample).transaction.id('n/a'),
+      serviceName: oc(sample).context.service.name(),
+      id: sample.transaction.id,
+      traceId: oc(sample).trace.id(),
```

Comment:
Like I said above, I think the interface should be changed:

Current interface:
```ts
export interface ITransactionGroup {
  name: string;
  serviceName?: string;
  id: string;
  traceId?: string;
  p95: number;
  averageResponseTime: number;
  transactionsPerMinute: number;
  impact: number;
  transactionType?: string;
}
```

I propose to change it to:
```ts
export interface ITransactionGroup {
  name: string;
  sample: {
    transactionType: string;
    serviceName: string;
    transactionId: string;
    traceId?: string;
  };
  p95: number;
  averageResponseTime: number;
  transactionsPerMinute: number;
  impact: number;  
}
```